service/application/src/main/java/com/solace/maas/ep/event/management/agent/config/SolaceConfiguration.java

```diff
@@ -62,10 +66,12 @@ public String getTopicPrefix() {
     @ConditionalOnMissingBean(EnableRtoCondition.class)
     @ConditionalOnProperty(name = "event-portal.gateway.messaging.standalone", havingValue = "false")
     public MessagingService messagingService() {
-        String clientName = "runtimeAgent-" + eventPortalProperties.getRuntimeAgentId();
+        String computedClientName = determineClientName();
+        String clientName = StringUtils.isEmpty(computedClientName) ?
+                "runtimeAgent-" + eventPortalProperties.getRuntimeAgentId() : computedClientName;
```

Comment:
Falls back on the ep core generated client name if EMA is having trouble determining its own client name based on host name. 
In other words, we are following precedence here
- EMA tries to generate its own client name 
- If it cannot, then it uses the one generated by ep core